test/ClearValue.lean

```diff
@@ -0,0 +1,32 @@
+import Mathlib.Tactic.Basic
+
+example : let x := 22; 0 ≤ x := by
+  intro x
+  clear_value x
+  fail_if_success clear_value x
+  exact Nat.zero_le _
+
+example : let x := 22; let y : Fin x := 0; y.1 < x := by
+  intro x y
+  fail_if_success clear_value x
+  clear_value y
+  clear_value x
+  fail_if_success clear_value x
+  fail_if_success clear_value y
+  exact y.2
+
+example : let x := 22; let y : Fin x := 0; y.1 < x := by
+  intro x y
+  fail_if_success clear_value x
```

Comment:
Why does this fail? I don't see how anything is dependent on `x` being `22`.

Reply:
It's the `0`, since to construct it it depends on `x = Nat.succ 21`.